addons/mrp/tests/test_order.py

```diff
@@ -5017,6 +5017,13 @@ def test_workcenter_with_resource_calendar_from_another_company(self):
         with self.assertRaises(UserError):
             self.workcenter_1.resource_calendar_id, = resource_calendar
 
+    def test_workorder_without_product(self):
+        mo_form = Form(self.env['mrp.production'])
+        with mo_form.workorder_ids.new() as wo:
+            wo.name = "Cutting"
+            wo.workcenter_id = self.workcenter_1
+        self.assertEqual(wo.name, "Cutting")
```

Comment:
Let's test a legit use case here
Set a finished product, save the MO avec make sure the WO is still ok